File: my-folder/0009-palindrome-number/solution.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        l=0
        xstr=str(x)
        r=len(xstr)-1
        while l<=r:
            if xstr[l]!=xstr[r]:
                return False
            l+=1
            r-=1
        return True
            


           
        

File: my-folder/0009-palindrome-number/solution.java
class Solution {
    public boolean isPalindrome(int x) {
        if(x < 0) return false;
        int original = x;
        int res = 0;
        while(x != 0){
            int digit = x % 10;

            if(res > Integer.MAX_VALUE / 10) return false;

            res = res * 10 + digit;
            x /= 10;
        }

        return res == original;
    }
}
